builtin/providers/cloudstack/resources.go

```diff
@@ -53,6 +56,9 @@ func retrieveUUID(cs *cloudstack.CloudStackClient, name, value string) (uuid str
 	case "network":
 		uuid, err = cs.Network.GetNetworkID(value)
 	case "zone":
+		if value == UnlimitedResourceID {
+			return value, nil
+		}
```

Comment:
If the `zone` ID may be -1, are there any other params that could maybe allow -1 as well? Because if that is the case, we should put this check above the switch statement I think...